v-next/hardhat/src/internal/cli/init/project-structure.ts

```diff
@@ -0,0 +1,55 @@
+import path from "node:path";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+import { findUp, readUtf8File } from "@nomicfoundation/hardhat-utils/fs";
+import { findClosestPackageJson } from "@nomicfoundation/hardhat-utils/package";
+
+const JS_CONFIG_FILENAME = "hardhat.config.js";
+const TS_CONFIG_FILENAME = "hardhat.config.ts";
+const CJS_CONFIG_FILENAME = "hardhat.config.cjs";
+const CTS_CONFIG_FILENAME = "hardhat.config.cts";
+
+export async function isCwdInsideProject() {
+  return (
+    (await findUp(TS_CONFIG_FILENAME)) !== undefined ||
+    (await findUp(CTS_CONFIG_FILENAME)) !== undefined ||
+    (await findUp(CJS_CONFIG_FILENAME)) !== undefined ||
+    (await findUp(JS_CONFIG_FILENAME)) !== undefined
+  );
+}
+
+export async function getUserConfigPath() {
+  const tsConfigPath = await findUp(TS_CONFIG_FILENAME);
+  if (tsConfigPath !== null) {
+    return tsConfigPath;
+  }
+
+  const ctsConfigPath = await findUp(CTS_CONFIG_FILENAME);
+  if (ctsConfigPath !== null) {
+    return ctsConfigPath;
+  }
+
+  const cjsConfigPath = await findUp(CJS_CONFIG_FILENAME);
+  if (cjsConfigPath !== null) {
+    return cjsConfigPath;
+  }
+
+  const jsConfigPath = await findUp(JS_CONFIG_FILENAME);
+  if (jsConfigPath !== null) {
+    return jsConfigPath;
+  }
+
+  throw new HardhatError(HardhatError.ERRORS.GENERAL.NOT_INSIDE_PROJECT);
+}
+
+export async function getRecommendedGitIgnore() {
+  const packageRoot = await getPackageRoot();
+  const gitIgnorePath = path.join(packageRoot, "recommended-gitignore.txt");
+
+  return readUtf8File(gitIgnorePath);
+}
+
+async function getPackageRoot(): Promise<string> {
+  const packageJsonPath = await findClosestPackageJson(import.meta.url);
+  return path.dirname(packageJsonPath);
+}
```

Comment:
I think it'd be better having the gitignore in a constant

Reply:
The original intention to have it as a file was that it was easy to find it in the root of the repo, but that's no longer the case, as we use a monorepo.

Reply:
Actually, we aren't using these functions, are we? If so, let's remove them. 

Note that combined with the other comment, that means removing this file entirely.

Reply:
I removed these 2 functions [here](https://github.com/NomicFoundation/hardhat/pull/5425/commits/a81b6fda5b98beb0e03fa5d2adfd48d0c8fe3828). For the other comment, I replied with a follow up question